File: AKNOVABROW/Services/SecurityService.cs
using System.Collections.Generic;
using System.Linq;

namespace AKNOVABROW.Services
{
    public class SecurityService
    {
        private readonly HashSet<string> maliciousPatterns = new()
        {
            "malware", "spyware", "trojan", "virus", "phishing",
            "scam", "fraud", "hack", "exploit", "ransomware",
            "keylogger", "rootkit", "botnet", "spam"
        };

        private readonly HashSet<string> blockedDomains = new()
        {
            "malicious-site.com",
            "phishing-scam.net",
            "fake-bank.com"
        };

        public bool IsSafe(string url)
        {
            url = url.ToLower();

            // Check blocked domains
            if (blockedDomains.Any(domain => url.Contains(domain)))
                return false;

            // Check malicious patterns
            if (maliciousPatterns.Any(pattern => url.Contains(pattern)))
                return false;

            return true;
        }

        public string GetThreatInfo(string url)
        {
            if (!IsSafe(url))
                return "⚠️ THREAT DETECTED: This site may contain malware or phishing content!";

            return "✅ Site appears safe";
        }
    }
}

File: aknovabrow_mobile/lib/services/security_service.dart
class SecurityService {
  final List<String> _maliciousPatterns = [
    'malware', 'spyware', 'trojan', 'virus', 'phishing',
    'scam', 'fraud', 'hack', 'exploit',
  ];

  final List<String> _blockedDomains = [
    'malicious-site.com',
    'phishing-scam.net',
  ];

  bool isSafe(String url) {
    final lowercaseUrl = url.toLowerCase();

    for (var domain in _blockedDomains) {
      if (lowercaseUrl.contains(domain)) return false;
    }

    for (var pattern in _maliciousPatterns) {
      if (lowercaseUrl.contains(pattern)) return false;
    }

    return true;
  }

  String getThreatInfo(String url) {
    if (!isSafe(url)) {
      return '⚠️ THREAT DETECTED: This site may contain malware or phishing content!';
    }
    return '✅ Site appears safe';
  }
}